src/isomorphic/classic/element/ReactElement.js

```diff
@@ -243,10 +243,19 @@ ReactElement.cloneElement = function(element, config, children) {
       key = '' + config.key;
     }
     // Remaining properties override existing props
+    var defaultProps = {};
```

Comment:
Thanks @jimfb . I usually have this allocation to avoid undefined checking, but not realize it has bad effect on gc. Updated my PR